components/basic-analyzers/src/main/java/org/datacleaner/beans/CompletenessAnalyzer.java

```diff
@@ -109,13 +110,9 @@ public String getName() {
     @Description("Optional additional values to add to output data streams")
     InputColumn<?>[] _additionalOutputValueColumns;
 
-    @Inject
-    @Provided
-    RowAnnotation _invalidRecords;
-
-    @Inject
-    @Provided
-    RowAnnotationFactory _annotationFactory;
+    // Do not inject the shared RowAnnotations, available rows are always needed.
+    final RowAnnotationFactory _annotationFactory = new InMemoryRowAnnotationFactory2();
```

Comment:
private ? :)

And wouldn't it be prettier to instantiate/assign this variable in the constructor - like the _invalidRecords variable?